core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1032,10 +1105,19 @@ public long createFile(AlluxioURI path, CreateFileOptions options)
       BlockInfoException, IOException, FileDoesNotExistException {
     Metrics.CREATE_FILES_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
-        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE)) {
-      mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
+        LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.WRITE);
+        MasterAuditContext auditContext =
+             createAuditContext("createFile", path, null,
+                 inodePath.peekInode(options.isRecursive()))) {
```

Comment:
I don't understand why the recursive flag is relevant to the audit log. HDFS doesn't even print out the permissions if the command fails, and prints out the target inode when it succeeds.

Therefore, wouldn't we only need to get the last existing inode in the path for any failure case, and the target inode for the success case?

(This question applies to anywhere `peekInode()` is used.)